src/library.js

```diff
@@ -3532,6 +3532,46 @@ LibraryManager.library = {
     }
   },
 
+#if SHRINK_LEVEL == 0
+  // A mirror copy of contents of wasmTable in JS side, to avoid relatively
+  // slow wasmTable.get() call. Only used when not compiling with -Os or -Oz.
+  _wasmTableMirror: [],
+
+  $wbind__deps: ['_wasmTableMirror'],
+  $wbind: function(funcPtr) {
+    var func = __wasmTableMirror[funcPtr];
+    if (!func) {
+      if (funcPtr >= __wasmTableMirror.length) __wasmTableMirror.length = funcPtr + 1;
+      __wasmTableMirror[funcPtr] = func = wasmTable.get(funcPtr);
+    }
+    return func;
```

Comment:
This cache is unsafe if the table is modified, which happens during dynamic linking, see `addFunction()` etc.

I think for now it would be fine to not get this speedup there. Can add a TODO, and to ifdef on line 3535 on RELOCATABLE as well.

Reply:
Check, resolved.